### PR TITLE
Or 1597 test the thnos repo for ecotone

### PR DESCRIPTION
**Pull request by rlgns98kr**
- I removed admin geth (we can remove and rerun devnet when we want to upgrade contracts). Because, the state file(genesis, allocs) is different from that made by `forge`
- I changed op-chain-ops. There was so many upgrade in Ecotone. So we have to check it.
- I updated op-binding. There was so many upgrade in Ecotone. Let's check this as well. I checked to create `hardhat binding` successfully.
- I updated the action to use github runner (ubuntu-latest-l: CPU:8, RAM:32). If you don't have to use ARM, you can use this runner in case of big task. It is charged the cost.

Thank you very much!

### CODE REVIEW

**Review by nguyenzung — COMMENTED** [review]
Excellent job, Steven!
You did a really hard task.

File: .github/workflows/e2e-tests.yml [hunk 1]
```diff
@@ -211,27 +58,6 @@ jobs:
           make cannon-prestate
           DEPLOYMENT_CONTEXT=devnetL1 make devnet-allocs
 
-      - name: Run e2e http test
-        run: OP_E2E_USE_HTTP=true OP_TESTLOG_DISABLE_COLOR=true OP_E2E_DISABLE_PARALLEL=false gotestsum --format=standard-verbose -- -timeout=25m -parallel=16 ./...
+      - name: Run ws e2e test
+        run: OP_TESTLOG_DISABLE_COLOR=true OP_E2E_CANNON_ENABLED=false OP_E2E_USE_FPAC=false OP_E2E_DISABLE_PARALLEL=false  gotestsum --format=standard-verbose -- -timeout=25m -parallel=16 -failfast ./...
```
[the remark this answers is not in the document]
Comment (boohyung) [reply .github/workflows/e2e-tests.yml:62]: OP_E2E_CANNON_ENABLED and OP_E2E_USE_FPAC is for fault proof. I think they are set to false.

File: pnpm-lock.yaml
Comment (boohyung) [inline pnpm-lock.yaml:1]: you don't need to change pnpm version. just remind. the lockfile version 9 is imcompatible with pnpm v8. so we should inform the software dependency of pnpm is ^9.0 in the guide or README.

File: op-bindings/predeploys/addresses.go [hunk 2]
```diff
@@ -84,38 +85,33 @@ var (
...
-	Predeploys["WNativeToken"] = &Predeploy{
-		Address:       WNativeTokenAddr,
-		ProxyDisabled: true,
-	}
+	Predeploys["WNativeToken"] = &Predeploy{Address: WNativeTokenAddr, ProxyDisabled: true}
 	Predeploys["L2CrossDomainMessenger"] = &Predeploy{Address: L2CrossDomainMessengerAddr}
 	Predeploys["L2StandardBridge"] = &Predeploy{Address: L2StandardBridgeAddr}
 	Predeploys["SequencerFeeVault"] = &Predeploy{Address: SequencerFeeVaultAddr}
 	Predeploys["OptimismMintableERC20Factory"] = &Predeploy{Address: OptimismMintableERC20FactoryAddr}
-	Predeploys["L1BlockNumber"] = &Predeploy{Address: L1BlockAddr}
```
Comment (boohyung) [inline op-bindings/predeploys/addresses.go:118]: My mistake. Thanks your checking.

File: op-chain-ops/genesis/layer_two.go [hunk 3]
```diff
@@ -32,7 +32,10 @@ func BuildL2Genesis(config *DeployConfig, l1StartBlock *types.Block) (*core.Gene
 		FundDevAccounts(db)
 	}
 
-	SetPrecompileBalances(db)
```
Comment (boohyung) [inline op-chain-ops/genesis/layer_two.go:35]: I cannot find any usage of `op-chain-ops/genesis/testdata/test-deploy-config-devnet-l1-set-precompiled.json`. We can remove `config.setPrecompileBalances` if we don't use it. What do you think about that?

File: .github/workflows/e2e-tests.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (rlgns98kr) [reply .github/workflows/e2e-tests.yml]: Thank you very much, Brave!
I removed OP_E2E_USE_FPAC=false!
We should enable OP_E2E_CANNON_ENABLED later. I think we can update when we apply FP system!

File: pnpm-lock.yaml
Comment (rlgns98kr) [reply pnpm-lock.yaml:1]: Yes, I wrote the document!
https://www.notion.so/tokamak/Checklist-for-Thanos-Ecotone-2eb8fecf3b7a4a45a0c0cc63833f9710
Thank you for checking!

File: op-chain-ops/genesis/layer_two.go (hunk 3 above)
Comment (rlgns98kr) [reply op-chain-ops/genesis/layer_two.go:35]: I added the option of setPrecompileBalances in the past to be able to select to use or not! So I added the option in the devnetL1-template! (I don't know why it is removed).
So, I think we can use option!

Comment (boohyung) [reply op-chain-ops/genesis/layer_two.go:35]: Ok. It doesn't matter. Originally it used 2 tests in `layer_two.go`. We don't use it now, but we might leave it in config and use it later. Thank you very much!

**Review by boohyung — APPROVED** [review]
+1
Thank you very much! You are great!!

**Pull request merged**